Index: profiles/wally/modules/third/imagecache_actions/imagecache_canvasactions.module
===================================================================
--- profiles/wally/modules/third/imagecache_actions/imagecache_canvasactions.module	(révision 64880)
+++ profiles/wally/modules/third/imagecache_actions/imagecache_canvasactions.module	(copie de travail)
@@ -130,8 +130,17 @@
 
   $x = imagecache_actions_keyword_filter($x, $image->info['width'], $layer->info['width']);
   $y = imagecache_actions_keyword_filter($y, $image->info['height'], $layer->info['height']);
-
-  return imageapi_toolkit_invoke('overlay', $image, array($layer, $x, $y, $alpha, $reverse));
+  
+  /**
+   *  Passage par référence posant problème à partir de PHP 5.3
+   *  @see http://fr.php.net/manual/fr/migration53.incompatible.php
+   *  @author tat@audaxis.com
+   *  @since 2011-08-17
+   */
+  if(version_compare(PHP_VERSION, '5.3.0') >= 0)
+    return imageapi_toolkit_invoke('overlay', $image, array(&$layer, $x, $y, $alpha, $reverse));
+  else
+    return imageapi_toolkit_invoke('overlay', $image, array($layer, $x, $y, $alpha, $reverse));
 }
 
 /**
