*** /index.php	2012-05-03 01:59:43.000000000 +0200
--- /index.php	2012-04-27 10:30:41.988514670 +0200
***************
*** 14,19 ****
--- 14,27 ----
  require_once './includes/bootstrap.inc';
  drupal_bootstrap(DRUPAL_BOOTSTRAP_FULL);
  
+ if(arg(0) == 'taxonomy' && arg(1) == 'term') {
+   $menu_item = menu_get_item();
+   if($menu_item['page_callback'] != 'page_manager_term_view') {
+     menu_rebuild();
+     //variable_set('site_offline',TRUE);
+   }
+ }
+ 
  $return = menu_execute_active_handler();
  
  // Menu status constants are integers; page content is a string.
