Index: profiles/wally/modules/third/ctools/page_manager/plugins/task_handlers/http_response.inc
===================================================================
--- profiles/wally/modules/third/ctools/page_manager/plugins/task_handlers/http_response.inc	(revision 63662)
+++ profiles/wally/modules/third/ctools/page_manager/plugins/task_handlers/http_response.inc	(working copy)
@@ -1,5 +1,4 @@
 <?php
-// $Id: http_response.inc,v 1.1.2.1 2010/08/06 18:23:11 merlinofchaos Exp $
 
 /**
  * @file
@@ -69,7 +68,7 @@
   'add finish' => 'settings',
 
   'required forms' => array(
-    'settings' => t('Panel settings'),
+    'settings' => t('HTTP Response settings'),
   ),
 
   'edit forms' => array(
@@ -116,9 +115,6 @@
   $task_name = page_manager_make_task_name($task['name'], $subtask['name']);
   $output = '';
 
-  $display = panels_panel_context_get_display($handler);
-
-  ctools_include('plugins', 'panels');
   ctools_include('context');
   ctools_include('context-task-handler');
 
@@ -130,19 +126,17 @@
   $args = array('handlers', $handler->name, 'actions');
   $rendered_operations = page_manager_render_operations($page, $operations, array(), array('class' => 'actions'), 'actions', $args);
 
-  $layout = panels_get_layout($display->layout);
-
   $plugin = page_manager_get_task_handler($handler->handler);
 
   $object = ctools_context_handler_get_task_object($task, $subtask, $handler);
-  $display->context = ctools_context_load_contexts($object, TRUE);
+  $context = ctools_context_load_contexts($object, TRUE);
 
-  $access = ctools_access_group_summary(!empty($handler->conf['access']) ? $handler->conf['access'] : array(), $display->context);
+  $access = ctools_access_group_summary(!empty($handler->conf['access']) ? $handler->conf['access'] : array(), $context);
   if ($access) {
-    $access = t('This panel will be selected if @conditions.', array('@conditions' => $access));
+    $access = t('This task will be selected if @conditions.', array('@conditions' => $access));
   }
   else {
-    $access = t('This panel will always be selected.');
+    $access = t('This task will always be selected.');
   }
 
   $rows = array();
@@ -187,13 +181,13 @@
   $info = theme('table', array(), $rows, array('class' => 'page-manager-handler-summary'));
 
   $title = $handler->conf['title'];
-  if ($title != t('Panel')) {
-    $title = t('Panel: @title', array('@title' => $title));
+  if ($title != t('HTTP Response')) {
+    $title = t('HTTP Response: @title', array('@title' => $title));
   }
 
   $output .= '<div class="clear-block">';
   if ($show_title) {
-  $output .= '<div class="handler-title clear-block">';
+    $output .= '<div class="handler-title clear-block">';
     $output .= '<div class="actions handler-actions">' . $rendered_operations['actions'] . '</div>';
     $output .= '<span class="title-label">' . $title . '</span>';
   }
@@ -206,7 +200,7 @@
 }
 
 /**
- * Set up a title for the panel based upon the selection rules.
+ * Set up a title for the handler based upon the selection rules.
  */
 function page_manager_http_response_title($handler, $task, $subtask) {
   if (isset($handler->conf['title'])) {
@@ -218,7 +212,7 @@
 }
 
 /**
- * General settings for the panel
+ * General settings for the handler
  */
 function page_manager_http_response_edit_settings(&$form, &$form_state) {
   $conf = $form_state['handler']->conf;
@@ -257,7 +251,6 @@
   // Go through arguments and see if they match.
   ctools_include('context');
   ctools_include('context-task-handler');
-  ctools_include('plugins', 'panels');
 
   // Add my contexts
   $contexts = ctools_context_handler_get_handler_contexts($base_contexts, $handler);
