Index: includes/bootstrap.inc
===================================================================
--- includes/bootstrap.inc	(révision 77683)
+++ includes/bootstrap.inc	(copie de travail)
@@ -729,6 +729,7 @@
 
   $result = $forced_cache
     && !drupal_session_started()
+    && isset($_SERVER['REQUEST_METHOD'])
     && ($_SERVER['REQUEST_METHOD'] == 'GET' || $_SERVER['REQUEST_METHOD'] == 'HEAD')
     && !count(drupal_get_messages(NULL, FALSE))
     && !drupal_is_cli();
@@ -1666,7 +1667,7 @@
   static $ip_address = NULL;
 
   if (!isset($ip_address)) {
-    $ip_address = $_SERVER['REMOTE_ADDR'];
+    $ip_address = isset($_SERVER['REMOTE_ADDR'])?$_SERVER['REMOTE_ADDR']:'0.0.0.0';
     
     // Only use parts of the X-Forwarded-For (XFF) header that have followed a trusted route.
     // Specifically, identify the leftmost IP address in the XFF header that is not one of ours.
