diff --git a/includes/database.mysql-common.inc b/includes/database.mysql-common.inc
index ead7f36..9c11ccd 100644
--- a/includes/database.mysql-common.inc
+++ b/includes/database.mysql-common.inc
@@ -30,6 +30,30 @@
  *   set of results will return a database query result resource. Other
  *   successful queries will return TRUE and failing queries will return FALSE.
  */
+
+// Include functions for automatic master/slave replication
+require_once './includes/database.mysql-replication.inc';
+
+function db_query($query) {
+  $args = func_get_args();
+  array_shift($args);
+  $unprefixed_query = $query;
+  
+  $query = db_prefix_tables($query);
+  if (isset($args[0]) and is_array($args[0])) {
+  // 'All arguments in one array' syntax
+    $args = $args[0];
+  }
+  _db_query_callback($args, TRUE);
+  
+  $query = preg_replace_callback(DB_QUERY_REGEXP, '_db_query_callback', $query);
+  // return _db_query($query);
+  
+  $run_on_slave = _db_is_slave_query($unprefixed_query, $query, $args);
+  
+  return _db_query($query, FALSE, $run_on_slave);
+}
+/*
 function db_query($query) {
   $args = func_get_args();
   array_shift($args);
@@ -41,7 +65,7 @@
   $query = preg_replace_callback(DB_QUERY_REGEXP, '_db_query_callback', $query);
   return _db_query($query);
 }
-
+*/
 /**
  * @ingroup schemaapi
  * @{